code/src/containers/forms/product-form/hooks/use-update-product/useUpdateProduct.ts

```diff
@@ -0,0 +1,36 @@
+import { useNavigate } from "react-router-dom";
+
+import routes from "@/constants/routes";
+import useSnackbar from "@/hooks/use-snackbar/useSnackbar";
+import { useUpdateProductMutation } from "@/store/api/productsApi";
+import { UpdateProductBody } from "@/types/product.types";
+
+const useUpdateProduct = () => {
+  const [updateProduct, requestState] = useUpdateProductMutation();
+
+  const { openSnackbarWithTimeout } = useSnackbar();
+
+  const navigate = useNavigate();
+
+  const onSubmit = async (body: UpdateProductBody) => {
+    try {
+      await updateProduct(body).unwrap();
+
+      openSnackbarWithTimeout({
+        variant: "success",
+        messageTranslationKey: "productForm.updation.success"
+      });
+
+      navigate(routes.dashboard.products.path);
```

Comment:
Should we have ability to return back to edit product page? If no, then we need to add `{ replace: true }` option

Reply:
I rethought it and now removed navigate. After successful update request, snackbar will be shown, but user won't be redirected. I guess this behaviour is common for update pages.